### ISSUE TITLE
USWDS - Testing: axe-tester not correctly reporting color contrast issues.

### ISSUE THREAD

**Issue by mejiaj** [uswds/uswds#3496]
## Description

Related to issue #3491. Our current axe testing didn't catch this contrast issue. We need to figure out why it didn't. As well as how we can improve our testing to catch these types of bugs.

## Steps to reproduce the issue

1. `git checkout jm-axe-contrast-check`
2. `gulp regression`
3. You should see an error like this
<details><summary>a11y error</summary>

```bash

2 failing

  1) fractal component
       "buttons"
         on small-desktop (412x732)
           has no aXe violations:
     Error: Found 1 aXe violations:
        [
  {
    "id": "color-contrast",
    "impact": "serious",
    "tags": [
      "cat.color",
      "wcag2aa",
      "wcag143"
    ],
    "description": "Ensures the contrast between foreground and background colors meets WCAG 2 AA contrast ratio thresholds",
    "help": "Elements must have sufficient color contrast",
    "helpUrl": "https://dequeuniversity.com/rules/axe/3.5/color-contrast?application=axeAPI",
    "nodes": [
      {
        "any": [
          {
            "id": "color-contrast",
            "data": {
              "fgColor": "#0076d6",
              "bgColor": "#005ea2",
              "contrastRatio": 1.46,
              "fontSize": "12.7pt (16.96px)",
              "fontWeight": "bold",
              "expectedContrastRatio": "4.5:1"
            },
            "relatedNodes": [
              {
                "html": "<button class=\"usa-button  text-blue\">Default</button>",
                "target": [
                  ".text-blue"
                ]
              }
            ],
            "impact": "serious",
            "message": "Element has insufficient color contrast of 1.46 (foreground color: #0076d6, background color: #005ea2, font size: 12.7pt (16.96px), font weight: bold). Expected contrast ratio of 4.5:1"
          }
        ],
        "all": [],
        "none": [],
        "impact": "serious",
        "html": "<button class=\"usa-button  text-blue\">Default</button>",
        "target": [
          ".text-blue"
        ],
        "failureSummary": "Fix any of the following:\n  Element has insufficient color contrast of 1.46 (foreground color: #0076d6, background color: #005ea2, font size: 12.7pt (16.96px), font weight: bold). Expected contrast ratio of 4.5:1"
      }
    ]
  }
]
        To debug these violations, install aXe at:
        https://www.deque.com/products/axe/
      at /Users/jamesmejia/web/uswds/spec/axe-tester.js:84:29

  2) fractal component
       "buttons"
         on large-desktop (1280x732)
           has no aXe violations:
     Error: Found 1 aXe violations:
        [
  {
    "id": "color-contrast",
    "impact": "serious",
    "tags": [
      "cat.color",
      "wcag2aa",
      "wcag143"
    ],
    "description": "Ensures the contrast between foreground and background colors meets WCAG 2 AA contrast ratio thresholds",
    "help": "Elements must have sufficient color contrast",
    "helpUrl": "https://dequeuniversity.com/rules/axe/3.5/color-contrast?application=axeAPI",
    "nodes": [
      {
        "any": [
          {
            "id": "color-contrast",
            "data": {
              "fgColor": "#0076d6",
              "bgColor": "#005ea2",
              "contrastRatio": 1.46,
              "fontSize": "12.7pt (16.96px)",
              "fontWeight": "bold",
              "expectedContrastRatio": "4.5:1"
            },
            "relatedNodes": [
              {
                "html": "<button class=\"usa-button  text-blue\">Default</button>",
                "target": [
                  ".text-blue"
                ]
              }
            ],
            "impact": "serious",
            "message": "Element has insufficient color contrast of 1.46 (foreground color: #0076d6, background color: #005ea2, font size: 12.7pt (16.96px), font weight: bold). Expected contrast ratio of 4.5:1"
          }
        ],
        "all": [],
        "none": [],
        "impact": "serious",
        "html": "<button class=\"usa-button  text-blue\">Default</button>",
        "target": [
          ".text-blue"
        ],
        "failureSummary": "Fix any of the following:\n  Element has insufficient color contrast of 1.46 (foreground color: #0076d6, background color: #005ea2, font size: 12.7pt (16.96px), font weight: bold). Expected contrast ratio of 4.5:1"
      }
    ]
  }
]
        To debug these violations, install aXe at:
        https://www.deque.com/products/axe/
      at /Users/jamesmejia/web/uswds/spec/axe-tester.js:84:29



[15:10:28] 'regression' errored after 18 s
[15:10:28] Error in plugin "gulp-spawn-mocha"
Message:
    Mocha exited with code 2
Details:
    domainEmitter: [object Object]
    domainThrown: false
```

</details>

---

It catches the intentional error, but not the **real** issue that's failling which is the hover state on accent hover button.

![image](https://user-images.githubusercontent.com/3385219/83450304-a0acb700-a41a-11ea-82a1-acb79a9f2096.png)

## Additional information [optional]

* Relevant research and support documents
* Screen shot images
* Notes
* And so on

**Comment by mejiaj** [comment]
Axe has this color contrast checker tool:
https://dequeuniversity.com/rules/axe/3.5/color-contrast?application=axeAPI

If you input the foreground `#FFFFFF` and background `#28A0CB`

You'll get this table:
<table><caption role="region" aria-live="polite" aria-atomic="true">Contrast Ratio = <span id="orRatio">3.00<span class="toOne"> : 1<span></span></span></span></caption><tbody><tr><th scope="col">WCAG Standard</th><th scope="col">Small Text</th><th scope="col" class="large-text-cls">Large Text</th></tr><tr><th scope="row">AA</th><td id="orSmallAA" class="">FAIL</td><td id="orLargeAA" class="pass">Pass</td></tr><tr><th scope="row">AAA</th><td id="orSmallAAA" class="">FAIL</td><td id="orLargeAAA" class="">FAIL</td></tr></tbody></table>

But it still won't show via axe-core.

**Comment by fafnirical** [comment]
Isn't this because the font is bold? https://www.w3.org/TR/WCAG20/#larger-scaledef defines large text as "18 point or 14 point bold". The text here is 1.06rem (~16.96px) with a `font-weight` of 700

**Comment by thisisdano** [comment]
@fafnirical Hm. Could be!

**Comment by thisisdano** [comment]
@mejiaj Is this still an issue?

**Comment by mejiaj** [comment]
@thisisdano no, we can close this. Not able to reproduce in the latest version of axe-core.
